File: backend-0541/app/Models/ijin_instruktur.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ijin_instruktur extends Model
{   
    use HasFactory;
    protected $table = 'ijin_instruktur';
    protected $primaryKey = 'id_ijin';
    protected $guarded = [];
     
    public function instruktur()
    {
        return $this->hasOne('App\Models\User\instruktur','id_instruktur','id_instruktur');
    }
    public function instrukturPengganti()
    {
        return $this->hasOne('App\Models\User\instruktur','id_instruktur','id_instruktur_pengganti');
    }
    public function jadwalHarian()
    {
        return $this->hasOne('App\Models\jadwal_harian','id_jadwal_harian' , 'id_jadwal_harian');
    }

}

File: mobile_app_gofit_0541/lib/Models/ijin_instruktur.dart
class IjinInstruktur{
  String? idIjin;
  String? idJadwalHarian;
  String? statusIjin;
  String? tanggalPengajuan;
  String? instrukturPengganti;

  IjinInstruktur({this.idIjin, this.idJadwalHarian, this.tanggalPengajuan, this.instrukturPengganti , this.statusIjin});

  factory IjinInstruktur.fromJson(Map<String, dynamic> json){
    return IjinInstruktur(
      idIjin: json['id_instruktur'].toString(),
      idJadwalHarian: json['id_jadwal_harian'].toString(), 
      tanggalPengajuan: json['tanggal_pengajuan'].toString(), 
      statusIjin: json['status_ijin'],
      instrukturPengganti: json['id_instruktur_pengganti'].toString(), 
      );
  }
  
}